src/test/goldens/polymer/lib/elements/array-selector.d.ts

```diff
@@ -59,8 +59,102 @@ declare namespace Polymer {
    * });
    * ```
    */
-  interface ArraySelector extends Polymer.Element {
+  class ArraySelector extends
+    Polymer.ArraySelectorMixin(
+    Polymer.Element) {
   }
+
+  /**
+   * Element mixin for recording dynamic associations between item paths in a
+   * master `items` array and a `selected` array such that path changes to the
+   * master array (at the host) element or elsewhere via data-binding) are
+   * correctly propagated to items in the selected array and vice-versa.
+   * 
+   * The `items` property accepts an array of user data, and via the
+   * `select(item)` and `deselect(item)` API, updates the `selected` property
+   * which may be bound to other parts of the application, and any changes to
+   * sub-fields of `selected` item(s) will be kept in sync with items in the
+   * `items` array.  When `multi` is false, `selected` is a property
+   * representing the last selected item.  When `multi` is true, `selected`
+   * is an array of multiply selected items.
+   */
+  function ArraySelectorMixin<T extends new(...args: any[]) => {}>(base: T): {
+    new(...args: any[]): {
+
+      /**
+       * An array containing items from which selection will be made.
+       */
+      items: any[]|null;
```

Comment:
Hmm, so this is defined @ https://github.com/Polymer/polymer/blob/master/lib/elements/array-selector.html as:
```
/**
 * An array containing items from which selection will be made.
 */
items: {
  type: Array,
},

```
Which analyzer reports as `type: 'Array'`. Since it doesn't have a default value, is the correct type for Analyzer to report actually `Array|undefined`? I guess I could add the `undefined` if I see that there is no default value (and that there was no explicit JSDoc `@type` annotation which should always override).

Added a TODO for now.